controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -493,142 +649,79 @@ private long findSegmentSplitsMerges(List<Segment> referenceSegmentsList, List<S
             } else {
                 return CompletableFuture.completedFuture(existing);
             }
-        }).thenCompose(record -> {
+        }).thenCompose(record -> getActiveEpochRecord(true).thenCompose(currentEpoch -> {
             if (!record.getObject().equals(EpochTransitionRecord.EMPTY)) {
                 // verify that it's the same as the supplied input (--> segments to be sealed
                 // and new ranges are identical). else throw scale conflict exception
-                if (!verifyRecordMatchesInput(segmentsToSeal, newRanges, record.getObject())) {
+                if (!RecordHelper.verifyRecordMatchesInput(segmentsToSeal, newRanges, false, record.getObject())) {
                     log.debug("scale conflict, another scale operation is ongoing");
                     throw new EpochTransitionOperationExceptions.ConflictException();
                 }
                 return CompletableFuture.completedFuture(record);
-            }
+            } else {
+                // check input is valid and satisfies preconditions
+                if (!RecordHelper.canScaleFor(segmentsToSeal, currentEpoch)) {
+                    // invalid input, log and ignore
```

Comment:
We are not logging and ignoring here, it is throwing, right? Also, could you justify why we would want to log and ignore here?

Reply:
No we want to throw and that is what we are doing. 
The method is incorrect.